rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,562 @@
+    - (-) Potential visual conflicts with the content
+- (+) Indentation of the content is clear.
+
+</details>
+
+## Doc-comment examples
+
+This section contains many examples for some different use cases we; Visualize them and emphasize the previously discussed characteristics.
+
+`somefile.nix`
+
+````nix
+{
+  /**
+  Documentation for the fundamental 'id' function
+  
+  # Examples
+  
+  ```
+  id 1
+  =>
+  1
+  ```
+  
+  # Type
+  
+  ```
+  id :: a -> a 
+  ```
+  
+  */
+  id = x: x;
+}
+````
+
+### Attribute bindings
+
+`somefile.nix`
+
+````nix
+  {
+    /**
+
+      mapAttrs is a well-known function 
+
+      # Examples
+
+      ```
+      # some code examples
+      ```
+    
+    */
+    mapAttrs = f: s: #...
+  }
+````
+
+### Indentation follows know nix's `''` behavior
+
+Indentation follows the know `''`-nix multiline strings behavior.
+Making usage more intuitive and and one point less to think about.
+
+````nix
+  {
+    /**
+    Line 1
+      Line 2
+    */
+    id = f: s: #...
+  }
+````
+->
+```markdown
+Line 1
+  Line 2
+```
+
+Or more advanced
+
+````nix
+  {
+    /**
+    # Title
+
+    Some foo bar
+
+    ## Subtitle
+
+    Some more 
+
+      indented 1
+
+        indented 2
+
+    */
+    id = f: s: #...
+  }
+````
+=>
+```markdown
+# Title
+
+Some foo bar
+
+## Subtitle
+
+Some more 
+
+  indented 1
+
+    indented 2
+
+```
+
+
+### NixOS Module documentation
+
+> Note: NixOS Modules also can produce documentation from their interface declarations. However, this does not include a generic description and usage examples.
+
+`myModule.nix`
+````nix
+/**
+  This is a custom module.
+  
+  It configures a systemd service.
+  
+  # Examples
+  
+  Different use case scenarios
+  how to use this module
+*/
+{config, ...}:
+{
+  config = f: s: f s;
+}
+````
+
+### Anonymous function documentation
+
+`function.nix`
+````nix
+/**
+  This is an anonymous function implementation.
+  
+  It does not have a name yet. 
+  Nevertheless, documentation can be right next to the implementation.
+  
+  The name gets assigned later.
+*/
+{a, b}:
+{
+  sum = a + b;
+}
+````
+
+### Anonymous expression documentation
+
+`exp.nix`
+````nix
+/**
+  This is an anonymous string.
+ 
+  It is the documentation for the "anonymous expression" use case.
+  
+  Although this example is relatively superficial, there might be use cases.
+*/
+"-p=libxml2/include/libxml2"
+````
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+## Changes the existing comments inside the code base
+
+This could be automated. (e.g., via codemod)
+
+Also, this affects only the `lib` directory and a few other places currently used to build the documentation.
+
+## Changes in the `nixdoc` tooling are required
+
+It remains an open question that needs to be tried;
+
+Can we still build the current Nixos manuals with the new standard?
+
+- We can wait to change everything.
+- Migration can happen in small steps.
```

Comment:
```suggestion
## Requires changes in existing tooling to produce documentation

nixdoc, the tool used to produce Nixpkgs library function documentation, would have to be modified to fit the new format.

This would be a small change, and anyway this RFC is agnostic to the tool used -- an entirely new tool could be developed, or the functionality be included as part of Nix.
```